flox/core.py

```diff
@@ -392,9 +412,9 @@ def offset_labels(labels: np.ndarray, ngroups: int) -> tuple[np.ndarray, int]:
 
 def factorize_(
     by: tuple,
-    axis,
+    axis: T_AxissOpt,  # TODO: What is axis doing in this function?
```

Comment:
if `len(axis) < by.ndim` (e.g. `by` is a 2D array, and `axis=-1`) then the label `'a'` along axis=1 is different from label `'a'` along axis=0. We handle this using "offset groups" below.

A lot of complicated groupbys can be handled by factorizing appropriately and some reshaping.

Reply:
I suppose I don't really get how `np.isscalar(axis)` implies all that. But I'm fine if you do. :) 
I'll remove the comment.

Reply:
yes lets remove. It's `np.isscalar(axis) and groupvar.ndim > 1`

Reply:
Done.